tests/test_vis.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from delphi.eval.compare_models import NextTokenStats, compare_models
+from delphi.eval.utils import load_text_from_dataset, load_validation_dataset, tokenize
+
+torch.set_grad_enabled(False)
+
+
+# define a pytest fixture for the model name
+@pytest.fixture
+def model_name():
+    return "roneneldan/TinyStories-1M"
+
+
+# define a pytest fixture for a default tokenizer using the model_name fixture
+@pytest.fixture
+def tokenizer(model_name):
+    return AutoTokenizer.from_pretrained(model_name)
+
+
+# define a pytest fixture for a default model using the model_name fixture
+@pytest.fixture
+def model(model_name):
+    return AutoModelForCausalLM.from_pretrained(model_name)
+
+
+# define a pytest fixture for the raw dataset
+@pytest.fixture
+def ds_txt():
+    return load_text_from_dataset(load_validation_dataset("tinystories-v2-clean"))[:100]
+
+
+# define a pytest fixture for the tokenized dataset
+@pytest.fixture
+def ds_tok(tokenizer, ds_txt):
+    return [tokenize(tokenizer, txt) for txt in ds_txt]
+
+
+# define a pytest fixture for a tokenized sample
+@pytest.fixture
+def sample_tok(ds_tok):
+    return ds_tok[0]
```

Comment:
I think you can safely replace this with a few lines in test_compare_models

Reply:
This is from aspirations of a lot more test coverage, to enable easy testing across parameter combinations and fixture re-use - but you're right, it is silly when there's just one test and no parameter combinations being tested.